Source/Scene/Primitive.js

```diff
@@ -1122,19 +1114,6 @@ define([
         vs = modifyForEncodedNormals(primitive, vs);
         var fs = appearance.getFragmentShaderSource();
 
-        // Create shadow cast program
-        var shadowsEnabled = defined(frameState.shadowMap) && frameState.shadowMap.enabled;
-        if (!defined(primitive._shadowCastSP) && shadowsEnabled && primitive._castShadows) {
```

Comment:
Is there an easy way to get this functionality back, where you can turn casting or receiving on and off? I think just modifying the commands' `castShadows` and `receiveShadows` would work.